x/uibc/ics20/ibc_module.go

```diff
@@ -33,6 +39,23 @@ func (am IBCModule) OnRecvPacket(
 	packet channeltypes.Packet,
 	relayer sdk.AccAddress,
 ) ibcexported.Acknowledgement {
+	// Allowing only registered token
+	var data ibctransfertypes.FungibleTokenPacketData
+	if err := ibctransfertypes.ModuleCdc.UnmarshalJSON(packet.GetData(), &data); err != nil {
+		ackErr := sdkerrors.ErrInvalidType.Wrap("cannot unmarshal ICS-20 transfer packet data")
+		return channeltypes.NewErrorAcknowledgement(ackErr)
+	}
+
+	_, denom, err := ibcutil.GetFundsFromPacket(packet)
+	if err != nil {
+		return channeltypes.NewErrorAcknowledgement(err)
+	}
+
+	_, err = am.lkeeper.GetTokenSettings(ctx, denom)
+	if err != nil && ltypes.ErrNotRegisteredToken.Is(err) {
+		return channeltypes.NewErrorAcknowledgement(err)
+	}
```

Comment:
if we move lines 43-57 to a helper function, then we can test that helper function.